exasol/ds/sandbox/lib/setup_ec2/run_setup_ec2.py

```diff
@@ -115,6 +119,45 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         next(self._lifecycle_generator)
 
 
+def _install_dependencies_and_report_status(
+    ec2_instance: Optional[EC2Instance],
+    key_file_location: Optional[str],
+    configuration: ConfigObject,
+    installer: AnsibleDependencyInstaller,
+) -> None:
+    if not ec2_instance.is_running:
+        LOG.error(
+            f"Error during startup of EC2 instance '{ec2_instance.id}', "
+            f"status {ec2_instance.state_name}."
+        )
+        return
+    host_name = ec2_instance.public_dns_name
+    if installer is not None:
```

Comment:
Do you really need this check?
1. `installer` is of type AnsibleDependencyInstaller, so theoretically it can't be none
2. Wouldn't it be better to raise an exception if `installer` is None, because in this case you do not install anything.

Reply:
Using an installer or not is actually optionally.
Please see the requirements in the related ticket #386.

Regarding the logic, i am happy for suggestions
* log error messages
* raise exceptions
* use return values